admin-dev/themes/new-theme/js/pages/product/carrier/CarrierSelector.vue

```diff
@@ -0,0 +1,186 @@
+<!--**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *-->
+<template>
+  <div class="carrier-selector">
+    <div
+      class="form-check form-check-radio form-radio"
+    >
+      <div
+        class="carrier-selector-line"
+        v-if="carrierChoices.length"
+      >
+        <checkboxes-dropdown
+          :choices="carrierChoices"
+          :label="getLabel"
+          :selected-choice-ids="this.selectedCarrierIds"
+          @selectChoice="addCarrier"
+          @unselectChoice="removeCarrier"
+          @click="showModifyAllShopsCheckbox"
+        />
+        <button
+          type="button"
+          v-if="this.selectedCarrierIds.length > 0"
+          class="btn btn-outline-secondary carrier-choices-clear"
+          @click="clearAllSelected"
+        >
+          <i class="material-icons">close</i>
+          {{ $t('allCarriers.label') }}
+        </button>
+      </div>
+    </div>
+    <div
+      v-if="selectedCarrierIds.length"
+      id="selected-carriers"
+    >
+      <span>
+        <ul>
+          <li
+            v-for="(selectedCarrier, key) in selectedCarriers"
+            :key="key"
+          >
+            {{ selectedCarrier.label }}<span v-if="key !== selectedCarriers.length -1 ">, </span>
+          </li>
+        </ul>
+      </span>
+    </div>
+    <div>
+      <div
+        v-if="modifyAllShopsVisible"
+        class="form-check form-check-radio form-checkbox"
+      >
+        <div
+          class="md-checkbox md-checkbox-inline"
+        >
+          <label class="required">
+            <input
```

Comment:
We should discuss this, I still don't understand why it has to be inside this component In a previous state of your PR it was handled like any other field and it worked fine for me locally 🤔